### DISCUSSION TITLE
Add a cmd_vel multiplexer

### DISCUSSION THREAD

**Discussion by jmm-slamcore** [iRobotEducation/create3_docs#321] [Ideas]
When using the Create for autonomous navigation you might want to override navigation commands with keyboard or joystick teleop. Publishing commands to the `/cmd_vel` topic from multiple sources can cause issues as there is no priority between publishers.

This is probably an advanced feature, but it could be nice to have an integrated cmd_vel multiplexer (which can be configured on the webserver using parameters) which subscribes to multiple topics like `/input/navigation`, `/input/teleop` etc and then sends the correct commands to the robot based on priorities set in the configuration.

There are existing packages that can do this (e.g. https://github.com/kobuki-base/cmd_vel_mux) but might be easier/useful to others if it is already integrated into the Create.

**Comment by alsora** [comment]
This is an interesting feature, but I'm not sure if we want to put it on the robot.
It looks to me that the multiplexer could run externally, together with the other applications that are producing velocity commands.
What would be the advantage of having it integrated into the robot?
The debug would be more challenging as the users wouldn't have full control over the this component.

**Reply by jmm-slamcore** [reply]
Yeah makes sense. I was thinking about it just for convenience so when the robot boots up the mux is already running but this can be done on the SBC side with a bring up script so probably doesn't make sense to have it on the robot as it would complicate things.

**Reply by alsora** [reply]
Yes, this probably would be a nice-to-have utility/example, similarly to how we have launch files for teleoperation https://github.com/iRobotEducation/create3_examples/tree/galactic/create3_teleop
